pkg/scheduler/scheduler.go

```diff
@@ -154,12 +159,51 @@ func WithBindTimeoutSeconds(bindTimeoutSeconds int64) Option {
 	}
 }
 
+// WithFrameworkRegistry sets the framework registry.
+func WithFrameworkRegistry(registry framework.Registry) Option {
+	return func(o *schedulerOptions) {
+		o.frameworkRegistry = registry
+	}
+}
+
+// WithFrameworkConfigProducerRegistry sets the framework plugin producer registry.
+func WithFrameworkConfigProducerRegistry(registry *frameworkplugins.ConfigProducerRegistry) Option {
+	return func(o *schedulerOptions) {
+		o.frameworkConfigProducerRegistry = registry
+	}
+}
+
+// WithFrameworkPlugins sets the plugins that the framework should be configured with.
+func WithFrameworkPlugins(plugins *kubeschedulerconfig.Plugins) Option {
+	return func(o *schedulerOptions) {
+		o.frameworkPlugins = plugins
+	}
+}
+
+// WithFrameworkPluginConfig sets the PluginConfig array that the framework should be configured with.
+func WithFrameworkPluginConfig(pluginConfig []kubeschedulerconfig.PluginConfig) Option {
+	return func(o *schedulerOptions) {
+		o.frameworkPluginConfig = pluginConfig
+	}
+}
+
 var defaultSchedulerOptions = schedulerOptions{
-	schedulerName:                  v1.DefaultSchedulerName,
-	hardPodAffinitySymmetricWeight: v1.DefaultHardPodAffinitySymmetricWeight,
-	disablePreemption:              false,
-	percentageOfNodesToScore:       schedulerapi.DefaultPercentageOfNodesToScore,
-	bindTimeoutSeconds:             BindTimeoutSeconds,
+	schedulerName:                   v1.DefaultSchedulerName,
+	hardPodAffinitySymmetricWeight:  v1.DefaultHardPodAffinitySymmetricWeight,
+	disablePreemption:               false,
+	percentageOfNodesToScore:        schedulerapi.DefaultPercentageOfNodesToScore,
+	bindTimeoutSeconds:              BindTimeoutSeconds,
+	frameworkRegistry:               frameworkplugins.NewDefaultRegistry(),
+	frameworkConfigProducerRegistry: frameworkplugins.NewDefaultConfigProducerRegistry(),
+	// The plugins and pluginConfig options are currently nil because we currently don't have
+	// "default" plugins. All plugins that we run through the framework currently come from two
+	// sources: 1) specified in component config, in which case those two options should be
+	// set using their corresponding With* functions, 2) predicate/priority-mapped plugins, which
+	// pluginConfigProducerRegistry contains a mapping for and produces their configurations.
+	// Once predicates and priorities are migrated to natively run as plugins, the below two parameters
```

Comment:
Suggest add an explicit TODO here, otherwise, we may overlook it.

Reply:
done.